### PR TITLE
Fix regressions

### PR DESCRIPTION
**Pull request by MaceWindu**
- fix mapping regression in evaluator
- fix binary operation type inference

### CODE REVIEW

**Comment by MaceWindu** [comment]
Another batch of fixes, will be merged tomorrow

File: Source/LinqToDB/SqlQuery/SqlBinaryExpressionHelper.cs [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+using System.Collections.Generic;
+
+namespace LinqToDB.SqlQuery
+{
+	internal static class SqlBinaryExpressionHelper
+	{
+		private static readonly Dictionary<(Type left, string operation, Type right), Type> _operationTypes = new()
+		{
+			// +
+			{ (typeof(byte), "+", typeof(byte   )), ((byte)1 + (byte   )1).GetType() },
```
Comment (viceroypenguin) [inline Source/LinqToDB/SqlQuery/SqlBinaryExpressionHelper.cs:11]: personally i prefer the following syntax:
```suggestion
			[(typeof(byte), "+", typeof(byte   )] = ((byte)1 + (byte   )1).GetType(),
```

makes it more obvious where the key and value are, especially in complex syntax like this.

**Review by viceroypenguin — APPROVED** [review]

**Pull request merged**